routes/api/contacts.js

```diff
@@ -1,25 +1,25 @@
-const express = require('express')
+const express = require('express');
+const router = express.Router();
+const { contacts: ctrl } = require('../../controlers');
 
-const router = express.Router()
+const { contactsSchema: schema } = require('../../schemas');
+const {
+  contactValidation: validation,
+  controlerWrapper,
+} = require('../../middlewares');
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get('/', controlerWrapper(ctrl.getAll));
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get('/:contactId', controlerWrapper(ctrl.getById));
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post('/', validation(schema), controlerWrapper(ctrl.add));
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.delete('/:contactId', controlerWrapper(ctrl.removeById));
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.put(
+  '/:contactId',
+  validation(schema),
```

Comment:
в тебе тут має бути інша схема валідації. Поля name, email i phone мають бути optional

Reply:
Розділив, але не розумію як тут валідацію номера телефону робити, писати регулярку?

Reply:
Так, писати регулярку

Reply:
Щось я неможу розібратися, якщо пишу регулярку згідно документаціі Joi:
   phone: Joi.string().pattern(new RegExp('/^(d{3})sd{3}-d{4}$/')).optional(), то підкреслює черврним і говорить Use a regular expression literal instead of the 'RegExp' constructor, 
   а коли RegExp прибераю то говорить: Error: regex must be a RegExp.
   Що не так?

Reply:
Виправив